### ISSUE TITLE
[apis] implement universal pagination for `createApi`

### ISSUE THREAD

**Issue by miguelrk** [netzo/netzo#45]
**TLDR**: Fragmented pagination styles are a challenge lots of developers have to cope with when integrating multiple APIs, as APIs use many different pagination strategies. [A unified pagination API](https://blog.apideck.com/abstracting-pagination-across-third-party-apis) that supports all types of pagination should be implemented by using cursors to encode all pagination data from the different strategies. We already have the API integrations under `netzo/integrations/apis`, so this should be implemented in the client at `netzo/integrations/create-api/pagination` so that each API integration can specify its defaults (similar to how auth already works).

**See [abstracting-pagination-across-third-party-apis](https://blog.apideck.com/abstracting-pagination-across-third-party-apis)**

`netzo/integrations/create-api/pagination.ts`:
```ts
// TODO: unified pagination across APIs by using cursors to encode pagination strategies
// see https://blog.apideck.com/abstracting-pagination-across-third-party-apis

export type PaginationOffsetOptions = {
  type: "offset"
  offsetProp: string // e.g. "offset", "skip"
  limitProp: string // e.g. "limit"
};

export type PaginationCursorOptions = {
  type: "cursor",
  cursorProp: string // e.g. "cursor", "after"
};

export type PaginationPageOptions = {
  type: "page",
  pageProp: string // e.g. "page", "pageNumber", "p"
};

export type PaginationLinkOptions = {
  type: "link",
  linkProp: string // e.g. "link", "next"
}

export type PaginationOptions = PaginationOffsetOptions | PaginationCursorOptions | PaginationPageOptions | PaginationLinkOptions;

/**
 * Pagination hook to mutate context based on pagination.type
 *
 * @see https://blog.apideck.com/abstracting-pagination-across-third-party-apis
 *
 * @param pagination {Pagination} - the pagination object containing the type its corresponding options
 * @param context {FetchContext} - the context object to mutate ({ request, options })
 */
export const pagination = async (
  pagination: Pagination = { type: "none" },
  { request: _, options }: FetchContext,
) => {
  const query = {} as Record<string, string>;
  const headers = {} as Record<string, string>;

  switch (options.type) {
    case "offset": return paginationOffset<T>(items, options);
    case "cursor": return paginationCursor<T>(items, options);
    case "page": return paginationPage<T>(items, options);
    case "link": return paginationLink<T>(items, options);
  }

  options.query = { ...options.query, ...query };
  options.headers = { ...options.headers, ...headers };
};
```

**Comment by miguelrk** [comment]
Closing as out of scope for the API clients (integrations). Pagination should be handled by the user for greater flexibility and maximum control.
